client/src/telemetry/metrics.yaml

```diff
@@ -0,0 +1,101 @@
+---
+$schema: moz://mozilla.org/schemas/glean/metrics/2-0-0
+
+page:
+  path:
+    type: string
+    lifetime: application
+    send_in_pings:
+      - page
+      - action
+    description: |
+      The URL path of the page that was viewed.
+    data_sensitivity:
+      - "web_activity"
+    bugs:
+      - "https://bugzilla.mozilla.org/show_bug.cgi?id=1782509"
+    data_reviews:
+      - "https://github.com/mdn/yari/pull/6813#issuecomment-1203705308"
+    notification_emails:
+      - mdn-team@mozilla.com
+    expires: never
+  referrer:
+    type: string
+    lifetime: application
+    send_in_pings:
+      - page
+      - action
+    description: |
+      The referring URL that linked to the page that was viewed.
+    data_sensitivity:
+      - "web_activity"
+    bugs:
+      - "https://bugzilla.mozilla.org/show_bug.cgi?id=1782509"
+    data_reviews:
+      - "https://github.com/mdn/yari/pull/6813#issuecomment-1203705308"
+    notification_emails:
+      - mdn-team@mozilla.com
+    expires: never
+  page_event:
+    type: event
+    lifetime: ping
+    send_in_pings:
+      - action
+    description: |
+      An event describing any non-click type page interaction  E.g
+       'Logged in', 'Offer hero viewed'
+    data_sensitivity:
+      - "web_activity"
+    bugs:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1782509
+    data_reviews:
+      - "https://github.com/mdn/yari/pull/6813#issuecomment-1203705308"
+    notification_emails:
+      - mdn-team@mozilla.com
+    expires: never
+    extra_keys:
```

Comment:
Rather than having one generic event and stuffing the important bits into the event extras, it is preferred to create individual events (i.e. one for each label), and then the "type" could be an extra for those specific events. This helps to ensure that data collections are properly data-reviewed, as this event would allow for adding new labels at any time with no clear indication that this is an expansion of data-collection and would require a data-review. This also makes the collection more deliberate, descriptive, and transparent to the user.

Reply:
You are correct, having an individual event for each button would be a little awkward in this case. I like your second option using the "source" of the click as an event extra. This pattern would allow you to add the "click" event to any new interactive items on the page in the future without having to undergo a fresh data-review (it's still collecting the same information, after all). Event extras are meant for capturing the context around the event, and the "source of a click" fits this nicely.

Reply:
FWIW It seems that essentially the `label` extra key of the `clicked` event further below (lines 86+) is essentially the "source" ("target" in DOM terminology) of the click.